### DISCUSSION TITLE
Is there a way to opt out decimal.js ?

### DISCUSSION THREAD

**Discussion by olup** [prisma/prisma#11892] [Q&A]
We are porting existing code to Prisma and being able to get regular number from our query would be great for compatibility. We are tied to the output of Prisma db pull in terms of schema so I'm wondering if there is a setting somewhere that we can tweak.

Many thanks.

**Comment by nurul3101** [comment]
Hey @olup :wave:

Unfortunately there is no settings available to tweak the output.
You can use [toNumber](https://mikemcl.github.io/decimal.js/#toNumber) method of Decimal.js to get regular number. 
A solution could be using [Middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) to run after every query and convert the field to be a regular number using the `toNumber` method.

Please let me know if this solves your issue or if you have other questions?

**Reply by olup** [reply]
Interesting! I am wondering about the typing though. Can I override this too ?

**Reply by nurul3101** [reply]
One possible solution could be that you can expect the field type to be `type: bigint | number`.

**Reply by olup** [reply]
I meant, is there a way for me to change the typings returned by prisma after my middleware so that generated type is not a `decimal` but the transformed `number` ? Other wise the returned objects from queries will still have a decimal field, even if actually being a number, am I wrong ?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
